### PR TITLE
ButtonIcon background & ButtonBase disabled update

### LINKED ISSUE #19590: Update Button, ButtonLink and ButtonIcon disabled, hover states
### Description

We want to update the opacity of all disabled buttons from `0.3` to `0.5`. Currently, the `ButtonLink` and `ButtonIcon` extensions use a `0.5` opacity for hover states. To avoid confusion and improve the visual distinction between button states we can update the components.

### Technical Details

- Update the disabled opacity of `ButtonPrimary` and `ButtonSecondary` variants from `0.3` to `0.5`.
- Modify the hover state of `ButtonLink` to include an underline for the active state (`primary/alternative`).
- Adjust the background color of `ButtonIcon` to use `background/default-hover`.

### Acceptance Criteria

- Ensure that the style updates mentioned in the Technical Details section have been completed.

### References

- [Notion page for Technical Sync](https://www.notion.so/Technical-Sync-db6759fdb0bf4d39bf91e10517cee5f8?pvs=4)

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

* Fixes #19590 

- Made a temporary disabled css variable, but we should consider adding to design tokens
- Update ButtonBase disabled opacity from 0.3 to 0.5 (this will update ButtonPrimary, ButtonSecondary, and ButtonLink)
- Update ButtonIcon hover and active state to have a background color and replace disabled opacity 
- Added ButtonIcon size Medium 😬

## Screenshots/Screencaps

See comments below for screenshots

## Manual Testing Steps
`yarn jest ui/components/component-library`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/component-library-components.scss [hunk 1]
```diff
@@ -4,6 +4,12 @@
 * This will help improve specificity and reduce the chance of
 * unintended overrides.
 **/
+
+// Temp CSS Variables
+:root {
+  --opacity-disabled: 0.5;
+}
+
```
Comment (georgewrmarshall) [inline ui/components/component-library/component-library-components.scss:12]: Ooh, interesting. Should we set aside some time to compare Sass vars vs CSS vars? Another option is to create a stylesheet for opacity similar to z-index.css.

Comment (georgewrmarshall) [reply ui/components/component-library/component-library-components.scss:12]: That makes sense. I'll create a ticket to add opacity tokens

File: ui/components/component-library/button-link/button-link.scss [hunk 2]
```diff
@@ -1,36 +1,46 @@
...
-    opacity: 0.5;
+  &:hover:not(&--disabled)::after {
+    content: '';
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    height: 2px;
+    background-color: currentColor;
   }
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/button-link/button-link.scss:10]: Wondering if we move the underline update to another PR then we can get the rest of these updates in?

File: ui/components/component-library/button-link/button-link.stories.js [hunk 3]
```diff
@@ -131,7 +131,7 @@ export const SizeStory = (args) => (
   <>
     <Box display={DISPLAY.FLEX} alignItems={AlignItems.flexStart} gap={4}>
       <ButtonLink {...args} size={Size.auto}>
-        Auto (default)
+        Auto (default)g
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-link/button-link.stories.js:134]:
```suggestion
        Auto (default)
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
All these updates look great! Because there are still conversations happening with ButtonLink does it make sense to move those to a separate PR and get the rest in? Left one suggestion on a typo

**Review by georgewrmarshall — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**